packages/generators/src/init-generator.ts

```diff
@@ -19,217 +20,57 @@ const { logger, getPackageManager } = utils;
-                } else {
-                    (this.configuration.config.webpackOptions.plugins as string[]).push(
-                        "new MiniCssExtractPlugin({ filename:'style.css' })",
-                    );
-                }
-
-                // Remove style-loader from the loader chain
-                ExtractUseProps.shift();
-
-                ExtractUseProps.unshift({
-                    loader: 'MiniCssExtractPlugin.loader',
-                });
-            }
 
-            // load CSS assets, with or without mini-css-extract-plugin
-            this.configuration.config.webpackOptions.module.rules.push({
-                test: regExpForStyles,
-                use: ExtractUseProps,
-            });
+            this.template = selectedTemplate;
         }
 
-        // webpack Dev Server
-        const { useDevServer } = await Confirm(
-            self,
-            'useDevServer',
-            'Do you want to use webpack-dev-server?',
-            true,
-            this.autoGenerateConfig,
-        );
-        if (useDevServer) {
-            this.dependencies.push('webpack-dev-server');
-            this.configuration.config.webpackOptions.devServer = {
-                open: true,
-                host: "'localhost'",
-            };
-        }
-
-        const { useHTMLPlugin } = await Confirm(
-            self,
-            'useHTMLPlugin',
-            'Do you want to simplify the creation of HTML files for your bundle?',
-            false,
-            this.autoGenerateConfig,
-        );
-
-        if (useHTMLPlugin) {
-            // Html webpack Plugin
-            this.dependencies.push('html-webpack-plugin');
-            const htmlWebpackDependency = 'html-webpack-plugin';
-            (this.configuration.config.topScope as string[]).push(
-                `const HtmlWebpackPlugin = require('${htmlWebpackDependency}')`,
-                '\n',
-                tooltip.html(),
-            );
-            (this.configuration.config.webpackOptions.plugins as string[]).push(`new HtmlWebpackPlugin({
-					template: 'index.html'
-				})`);
-        }
-
-        const { useWorkboxPlugin } = await Confirm(
-            self,
-            'useWorkboxPlugin',
-            'Do you want to add PWA support?',
-            true,
-            this.autoGenerateConfig,
-        );
-        // webpack Dev Server
-        if (useWorkboxPlugin) {
-            this.configuration.config.topScope.push("const WorkboxWebpackPlugin = require('workbox-webpack-plugin');", '\n');
-            this.dependencies.push('workbox-webpack-plugin');
-            (this.configuration.config.webpackOptions.plugins as string[]).push(`new WorkboxWebpackPlugin.GenerateSW({
-				swDest: 'sw.js',
-				clientsClaim: true,
-				skipWaiting: false,
-			})`);
-        }
-
-        this.configuration.config.webpackOptions.mode = this.autoGenerateConfig ? "'production'" : "'development'";
+        await handlers[this.template].questions(this, Question);
```

Comment:
Isn't the `Question` argument extraneous? The required prompts can be imported directly within [packages/generators/src/handlers/default.ts](https://github.com/webpack/webpack-cli/pull/2429/files#diff-50d01c8f19701cb38c78b1743da73ab584d818f35d9e0db63c9310b17a522e1e)

```js
import { Confirm, Input, List } from './utils/scaffold-utils';
```

Reply:
Not exactly, I am making `handlers` independent of the `webpack-cli` code. So that we can implement support 3rd party templates as well in future without any issues.